refactor: Add internal token middleware and enhance coins

amirxhero opened the pull request:
Introduces a new middleware to set and validate an internal token for non-API and /api/coins routes, improving request authentication. The coins.get API handler now checks the request's origin and referer against allowed hosts, further strengthening access control.

amirxhero committed:
feat (coins) Add coins API proxy and update coin fetch endpoint

Introduces a new server API route at /api/coins that proxies requests to /api/v2/game/coins with a secret key from runtime config. Updates RandomCoin.vue to use the new endpoint and refactors nuxt.config.js for improved readability and configuration.

amirxhero committed:
refactor (components) : Add internal token middleware and enhance coins API security

Introduces a new middleware to set and validate an internal token for non-API and /api/coins routes, improving request authentication. The coins.get API handler now checks the request's origin and referer against allowed hosts, further strengthening access control.

amirxhero committed:
refactor (components) : Add route-based conditional rendering to RandomCoin

RandomCoin component now checks the current route name and avoids rendering on excluded routes such as 'search' or any route with a 'search-' prefix. This prevents the coin animation from appearing on specific pages where it is not desired.

sanaderi commented on components/common/RandomCoin.vue:
```diff
@@ -41,14 +42,23 @@ import successSound from '/assets/sounds/success.mp3'
+const shouldRenderRandomCoin = computed(() => {
+  const name = route.name ? String(route.name) : ''
+  return !excludedRouteNames.some(
+    n => name === n || name.startsWith(`${n}-`),
```
Just exclude a specific route address; this condition causes the coin not to appear, for example, on the school-id-slug route

sanaderi commented on components/common/RandomCoin.vue:
```diff
@@ -41,14 +42,23 @@ import successSound from '/assets/sounds/success.mp3'
 
 let scrollHandler = null
 
 const { data: coinsResponse, refresh: refreshCoins } = await useAsyncData(
```
Don't call the coins API on pages in the excluded list

amirxhero committed:
refactor (components) : Refactor coin game authentication and API usage

Updated coin game logic to require authentication before coin collection, switched API endpoints to v2, and improved coin data handling. Added support for preventing automatic redirects after login via query parameter. Cleaned up API service header merging and removed unused server files.

amirxhero committed:
refactor (components) : Refactor RandomCoin rendering logic for route checks

Simplifies route exclusion logic and ensures coin data is only fetched and displayed when the current route is not excluded. Adds async initialization for coin data if needed and guards coin display and refresh logic based on route.

amirxhero committed:
Refactor RandomCoin API key usage and update config

Replaced hardcoded API key in RandomCoin.vue with a runtime config value for improved security and flexibility. Added randomCoinApiKey to nuxt.config.js public runtime config and updated code style to use consistent double quotes.

amirxhero committed:
refactor (components) : Refactor RandomCoin data fetching and update exclusions

Simplifies the coin data fetching logic in RandomCoin.vue by removing asyncData and refresh logic in favor of a direct API call. Updates the list of excluded route names to include 'game-castle' and 'game-car-racing'. Also removes the v-once directive from lazy-common-random-coin in the default layout to allow reactivity.
